api/v1alpha1/ndb_api_helpers.go

```diff
@@ -155,43 +157,147 @@ func GetNoneTimeMachineSLA(ctx context.Context, ndbclient *ndbclient.NDBClient)
-	})
-	storageProfiles := util.Filter(genericProfiles, func(p ProfileResponse) bool { return p.Type == PROFILE_TYPE_STORAGE })
-	// Specifying the usage of single instance topoplogy
-	softwareProfiles := util.Filter(dbEngineSpecificProfiles, func(p ProfileResponse) bool { return p.Type == PROFILE_TYPE_SOFTWARE && p.Topology == TOPOLOGY_SINGLE })
-	networkProfiles := util.Filter(dbEngineSpecificProfiles, func(p ProfileResponse) bool { return p.Type == PROFILE_TYPE_NETWORK })
-	dbParamProfiles := util.Filter(dbEngineSpecificProfiles, func(p ProfileResponse) bool { return p.Type == PROFILE_TYPE_DATABASE_PARAMETER })
-
-	// Some profile not found, return an error
-	if len(computeProfiles) == 0 || len(softwareProfiles) == 0 || len(storageProfiles) == 0 || len(networkProfiles) == 0 || len(dbParamProfiles) == 0 {
-		err = errors.New("oob profile: one or more OOB profile(s) were not found")
+	/*
+		1. if both name & id not provided => resolve the OOB profile
+		2. else if both name & id are provided => resolve by both & ensure that both resolved profiles are match
+		3. else if only id provided => resolve by id
+		4. else if only name provided => resolve by name
+		5. else => throw an error
+	*/
+	if !isNameProvided && !isIdProvided { // OOB
+		log.Info("Attempting to resolve the OOB profile, no id or name provided in the spec", "Profile", pType)
+		oobProfile, err := util.FindFirst(allProfiles, filter)
+
+		if err != nil {
+			log.Error(err, "Error resolving OOB Profile", "type", pType)
+			return ProfileResponse{}, fmt.Errorf("no OOB profile found of type=%v", pType)
+		}
+		return oobProfile, nil
+
+	} else if isNameProvided && isIdProvided { // verify that both resolved profiles (by id and name) are one and the same
+		if !reflect.DeepEqual(profileById, profileByName) {
+			log.Error(err, "profile matching both the given name & id does not exist. Retry with correct inputs")
+			return ProfileResponse{}, fmt.Errorf("profiles returned by id & name resolve to different profiles")
+		}
+		return profileById, nil
+
+	} else if isIdProvided {
+		return profileById, nil
+
+	} else if isNameProvided {
+		return profileByName, nil
+	}
+
+	return ProfileResponse{}, fmt.Errorf("could not resolve the profile by Name or Id, err=%v", err)
+}
+
+// TODO: Once the database_types refactoring is over, move out below methods to  profile_helper.go
+var ComputeOOBProfileResolver = func(p ProfileResponse) bool {
+	return p.Type == PROFILE_TYPE_COMPUTE &&
+		strings.EqualFold(p.Name, PROFILE_DEFAULT_OOB_SMALL_COMPUTE)
+}
+
+var SoftwareOOBProfileResolverForSingleInstance = func(p ProfileResponse) bool {
+	return p.Type == PROFILE_TYPE_SOFTWARE && p.Topology == TOPOLOGY_SINGLE
+}
+
+var NetworkOOBProfileResolver = func(p ProfileResponse) bool {
+	return p.Type == PROFILE_TYPE_NETWORK
+}
+
+var DbParamOOBProfileResolver = func(p ProfileResponse) bool {
+	return p.Type == PROFILE_TYPE_DATABASE_PARAMETER
+}
```

Comment:
I think your changes to check for systemProfile got lost in the rebase.. 🫢

Reply:
Fixed!